### ISSUE TITLE
Array internal pointer functions not giving warnings

### ISSUE THREAD

**Issue by iltar** [kalessil/phpinspectionsea#667]
When using functions such as reset, current and end (possibly more), they will return `false` when the array is empty. In case of these functions, it's unsafe to rely on direct usability of those functions and should always be checked if the result was not false If the array was guaranteed to have contents, either by a previous check or knowledge of contents, it shouldn't be necessary to verify again.

In the following examples, you would try to access an array key on `false` or a property on a non-object.
```php
$a = [];
$b = reset($a);
$b['foo'];

$c = [];
reset($c)['foo'];
current($c)['foo'];
end($c)['foo'];

/** @var $d \stdClass[] */
$d = [];
reset($d)->foo;
```

**Comment by kalessil** [comment]
@iltar : I tried introducing the inspection in an early version of the plugin, feedback was 100% negative unfortunately.

**Comment by funivan** [comment]
Maybe it would be better to write the wrapper for `reset`/`end` function.
For example when we get `false` -> throw new exception 
For legacy projects it would be very annoying to check `if false` every time))

**Comment by kalessil** [comment]
`very annoying to check if false every time` is exact feedback I got xD

**Comment by iltar** [comment]
Do you want to write proper code or do you want to write as little code as possible and accept failures all the time? I prefer the first :smile:

**Comment by funivan** [comment]
@iltar you are totally correct. The first option is the best
